### PR TITLE
fix: update navigation states and mobile menus

### PR DESCRIPTION
**Pull request by austin94**
## Description

This PR includes:

- Updates mobile styling for the global Header and Sidebar layouts
- Consolidates styling and states for all Header and Sidebar navigation elements
- Updates Header shadow to use larger box-shadow
- Increases touch-target of sidebar TOC
- Small bug fix to help with valid TOC headings while scrolling

Will be based off of `master` once #20 is merged

## Checklist

- [x] :ok_hand: website updates are Garden Designer approved (add the designer as a reviewer)
- [ ] ~:black_nib: copy updates are approved (add the content strategist as a reviewer)~
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :zap: audited via [web.dev](https://web.dev/measure/) for performance and SEO
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: src/layouts/Root/components/StyledNavigationLink.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  color: ${p => p.theme.colors.foreground};
+
+  &:focus,
+  &:hover,
+  &:active {
+    text-decoration: none;
+    color: inherit;
+  }
+
+  &.is-active {
```
Comment (jzempel) [inline src/layouts/Root/components/StyledNavigationLink.tsx:24]: Perhaps `is-current` to differentiate from `:active`? Avoids future confusion. Also aligns with https://github.com/zendeskgarden/react-components/blob/master/packages/chrome/src/styled/nav/StyledNavItem.ts.

File: src/layouts/Titled/components/TOC.tsx [hunk 2]
```diff
@@ -116,7 +125,14 @@ export const TOC: React.FC<{ data: IHeading[] }> = ({ data }) => {
       </li>
       {data.map(heading => (
         <StyledTocItem key={heading.url} isActive={activeHeadingUrl === heading.url}>
```
Comment (jzempel) [inline src/layouts/Titled/components/TOC.tsx:127]: Suggest rename `isActive` -> `isCurrent`.

Comment (austin94) [reply src/layouts/Titled/components/TOC.tsx:127]: Updated.

File: src/layouts/Root/components/StyledNavigationLink.tsx (hunk 1 above)
Comment (austin94) [reply src/layouts/Root/components/StyledNavigationLink.tsx]: Updated.

**Pull request merged**